material-ui/material-ui.d.ts

```diff
@@ -7228,7 +7228,7 @@ declare module 'material-ui/utils/withWidth' {
       mediumWidth?: number;
       resizeInterval?: number;
   }
-  export default function withWidth(options?: Options): __React.ComponentClass<any>
+  export default function withWidth<C extends __React.ComponentClass<any> | __React.StatelessComponent<any>>(options?: Options): (component: C) => C;
```

Comment:
@KostyaEsmukov very good point indeed. @mhegazy do you have an idea how we can address this scenario as well?